apps/core/app/[locale]/(default)/login/_components/change-password-form.tsx

```diff
@@ -73,18 +78,56 @@ export const ChangePasswordForm = ({ customerId, customerToken }: Props) => {
     messageText = t('successMessage');
   }
 
-  const handleNewPasswordChange = (e: ChangeEvent<HTMLInputElement>) =>
-    setNewPasssword(e.target.value);
+  const handleCurrentPasswordChange = (e: ChangeEvent<HTMLInputElement>) =>
+    setIsCurrentPasswordValid(!e.target.validity.valueMissing);
+  const handleNewPasswordChange = (e: ChangeEvent<HTMLInputElement>) => {
+    let formData: FormData | null = null;
+    let currentPasswordValue: FormDataEntryValue | null = null;
+    let isValid = true;
+    const newPasswordValue = e.target.value;
+
+    if (e.target.form) {
+      formData = new FormData(e.target.form);
+    }
```

Comment:
can we have a case when we have no `form` and formData is `null`?

Reply:
Yeah, it's a good question. `e.target.form` possibly can be `null` but your question brought  me up to a small refactoring  -I've simplified code and removed checking for `FormData` is `null` on a second step. Now it's done like:

```
if (e.target.form) {
   currentPasswordValue = new FormData(e.target.form).get('current-password');
}
```